packages/app/src/cli/models/extensions/specifications/payments_app_extension.ts

```diff
@@ -0,0 +1,49 @@
+import {createExtensionSpecification} from '../specification.js'
+import {BaseSchema} from '../schemas.js'
+import {zod} from '@shopify/cli-kit/node/schema'
+
+export type PaymentsAppExtensionConfigType = zod.infer<typeof PaymentsAppExtensionSchema>
+const PaymentsAppExtensionSchema = BaseSchema.extend({
+  payment_session_url: zod.string().url(),
+  refund_session_url: zod.string().url(),
+  capture_session_url: zod.string().url(),
+  void_session_url: zod.string().url(),
+  confirmation_callback_url: zod.string().url().optional(),
+  merchant_label: zod.string(),
+  supported_countries: zod.array(zod.string()),
+  supported_payment_methods: zod.array(zod.string()),
+  test_mode_available: zod.boolean(),
```

Comment:
Isn't this missing a lot of fields of payment apps (e.g. balance url for gift cards, gift card type, supports oversell protection, etc)? Or does this schema only include the common ones?

Reply:
I thought we already decided to have 1 single generic specification for all payment extensions no?

Reply:
Gotcha, if there are fields that are common to all they should be outside of the union though 👌